### PR TITLE
GLSL Goodies

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**

### CODE REVIEW

File: include/irr/builtin/material_compiler/glsl/common.glsl [hunk 1]
```diff
@@ -0,0 +1,528 @@
+#version 430 core
```
Comment (devshgraphicsprogramming) [inline include/irr/builtin/material_compiler/glsl/common.glsl:1]: rename this header, its not common to all material compilers, only one particular backend (or split out the common and non common part).

For example virtual texturing is only really a part of a backend that does not use the descriptor indexing Vulkan capability.

also path is wrong, should be `irr/builtin/glsl/material_compiler` I think

**Comment by devshgraphicsprogramming** [comment]
If I merge this, will the denoiser break? (it needs a functional mitsuba loader to load camera XMLs)

File: include/irr/builtin/material_compiler/glsl/common.glsl (hunk 1 above)
Comment (Crisspl) [reply include/irr/builtin/material_compiler/glsl/common.glsl:1]: > its not common to all material compilers, only one particular backend

thats why its in subdir for glsl backend

> also path is wrong, should be irr/builtin/glsl/material_compiler I think

that would imply scattering material_compiler sources over language/backend-based directories like glsl/material_compiler, backendX/material_compiler, backendY/material_compiler etc

Comment (devshgraphicsprogramming) [reply include/irr/builtin/material_compiler/glsl/common.glsl:1]: hmm ok makes sense, buut there are many possible backends for GLSL:
- GLSL separate pipeline/shader per BSDF (no multi draw indirect)
- GLSL with virtual geometry and virtual textures (no descriptor indexing)
- GLSL with bindless (descriptor indexing)

plus there is the question if the rasterizer backend will want/need the sampling functons (for stochastic shadows)

**Comment by Crisspl** [comment]
> If I merge this, will the denoiser break? (it needs a functional mitsuba loader to load camera XMLs)

no, it loads metadata just fine, provided shaders wont work obviously but thats it

**Comment by devshgraphicsprogramming** [comment]
ok fix beckmann and ggx, then I'll merge ;)

File: include/irr/builtin/material_compiler/glsl/common.glsl (hunk 1 above)
Comment (devshgraphicsprogramming) [reply include/irr/builtin/material_compiler/glsl/common.glsl:1]: @Crisspl take not of this comment, I'm closing the PR

**Pull request closed without merging**